### ISSUE TITLE
Problem with 6.17.0 with default value using overloaded "|" operator

### ISSUE THREAD

**Issue by labodj** [bblanchon/ArduinoJson#1411]
Hi,
since 6.17.0 this code is not working, it does not compile because it's complaining about the ambiguity of | overloaded operator.

```c++
#include <Arduino.h>
#include <ArduinoJson.h>


void setup()
{
}

void loop()
{
  const size_t capacity = JSON_OBJECT_SIZE(1) + 20;
  DynamicJsonDocument doc(capacity);
  const char *json = "{\"sensor\":\"gps\"}";
  deserializeJson(doc, json);
  const char *test = "test";
  const char *sensor = doc["sensor"] | test; // "gps"
}
```

Commenting lines 30-37 in /src/ArduinoJson/Variant/VariantOperators.hpp solve this problem, but can cause other unexpected ones.

**Comment by bblanchon** [comment]
Hi @labodj,

Thank you very much for reporting this bug.
The fix is available on the "6.x" branch.
I'll publish it in the next revision in a few days.

Best regards,
Benoit

**Comment by bblanchon** [comment]
The fix was published in [ArduinoJson 6.17.1](https://github.com/bblanchon/ArduinoJson/releases/tag/v6.17.1)
